src/selectors/per-thread/markers.js

```diff
@@ -605,11 +611,110 @@ export function getMarkerSelectorsPerThread(
     }
   );
 
+  type MarkerTrackSelectors = $ReturnType<typeof _createMarkerTrackSelectors>;
+  const _markerTrackSelectors = {};
+  const getMarkerTrackSelectors = (name: string): MarkerTrackSelectors => {
+    let selectors = _markerTrackSelectors[name];
+    if (!selectors) {
+      selectors = _createMarkerTrackSelectors(name);
+      _markerTrackSelectors[name] = selectors;
+    }
+    return selectors;
+  };
+
+  /**
+   * This function creates selectors for each of the trackable markers of a thread. The type
+   * signature of each selector is defined in the function body, and inferred in the return
+   * type of the function.
+   */
+  function _createMarkerTrackSelectors(name: string) {
+    const _getMarkerSchema: Selector<MarkerSchema> = createSelector(
+      ProfileSelectors.getMarkerSchemaByName,
+      (schemaByName) => schemaByName[name]
+    );
+
+    const _getMarkerTrackConfig: Selector<MarkerTrackConfig> = createSelector(
+      _getMarkerSchema,
+      (schema) => {
+        if (schema.trackConfig === undefined) {
+          throw new Error('No trackConfig for marker ' + name);
+        }
+        return schema.trackConfig;
+      }
+    );
+
+    const _getTrackKeys: Selector<string[]> = createSelector(
+      _getMarkerTrackConfig,
+      (trackConfig) => trackConfig.lines.map((lineConfig) => lineConfig.key)
+    );
+
+    const getCollectedCustomMarkerSamples: Selector<CollectedCustomMarkerSamples> =
+      createSelector(
+        getFullMarkerList,
+        threadSelectors.getStringTable,
+        _getTrackKeys,
+        (fullMarkerList, stringTable, trackKeys) => {
+          const filteredIndexes = fullMarkerList
+            .map((n, i) => [n, i])
+            .filter((t) => t[0].name === name)
+            .map((t) => t[1]);
+          const markers = filteredIndexes.map((i) => fullMarkerList[i]);
+          const numbersPerLine: number[][] = trackKeys.map((key) =>
+            markers.map((marker) => {
+              const markerPayload = marker.data;
+              if (
+                !(markerPayload instanceof Object) ||
+                !(key in markerPayload)
+              ) {
+                throw new Error('Missing ' + key + ' in marker ' + name);
+              }
+              return markerPayload[key];
+            })
+          );
+          const minNumber = Math.min(
+            ...numbersPerLine.map((x) => Math.min(...x))
+          );
+          const maxNumber = Math.max(
+            ...numbersPerLine.map((x) => Math.max(...x))
+          );
+          const length = numbersPerLine.length;
+          return {
+            minNumber,
+            maxNumber,
+            markers,
+            time: markers.map((marker) => marker.start),
+            numbersPerLine,
+            indexes: filteredIndexes,
+            length,
```

Comment:
I think you're specifying this because `getInclusiveSampleIndexRangeForSelection` uses it, but I don't think this has the right value. Indeed it needs to be the length of the time array.

Maybe `getInclusiveSampleIndexRangeForSelection` could be changed so that it does that: take the length of the time array instead of the separate property, and then you don't even need to add a `length` property here...